docs/content/intro/tutorial-preview-environments.md

```diff
@@ -0,0 +1,424 @@
+
+## Config Brigade Project
+
+Let's create new Brigade Project for our `brigade-tutorial-config` repository.
+
+```console
+$ brig project create --namespace brigade
+? Project Name brigadecore/brigade-tutorial-config
+? Full repository name github.com/brigadecore/brigade-tutorial-config
+? Clone URL https://github.com/brigadecore/brigade-tutorial-config
+```
+
+## Create Preview Environment Pipeline
+
+New environments will be created by executing brigade script via `brig` cli. 
+
+![Create Environment Pipeline](https://docs.brigade.sh/img/preview-environments-create.png)
+
+## Create Namespace
+
+Let's start by adding a new `brigade.js` script to the root of your repository. For now it will be responsible for creation of a new Kubernetes namespace.
+
+```js
+
+// brigade.js
+
+const { events } = require("@brigadecore/brigadier");
+const kubernetes = require("@kubernetes/client-node");
+
+const kubeConfig = new kubernetes.KubeConfig();
+kubeConfig.loadFromDefault();
+
+const k8sCoreClient = kubeConfig.makeApiClient(kubernetes.Core_v1Api);
+
+const protectedEnvironment = namespaceName => {
+  const protectedNamespaces = ["default", "kube-public", "kube-system", "brigade"];
+
+  if (protectedNamespaces.includes(namespaceName)) {
+    return true;
+  }
+  return false;
+};
+
+const createNamespace = async namespaceName => {
+  const existingNamespace = await k8sCoreClient.listNamespace(
+    true,
+    "",
+    `metadata.name=${namespaceName}`,
+  );
+
+  if (existingNamespace.body.items.length) {
+    console.log(`Namespace "${namespaceName}" already exists`);
+    return;
+  }
+
+  const namespace = new kubernetes.V1Namespace();
+  namespace.metadata = new kubernetes.V1ObjectMeta();
+  namespace.metadata.name = namespaceName;
+
+  await k8sCoreClient.createNamespace(namespace);
+};
+
+const provisionEnvironment = async (environmentName, projects) => {
+  await createNamespace(environmentName);
+};
+
+events.on("exec", event => {
+    const payload = JSON.parse(event.payload);
+    const { name } = payload;
+
+    if (!name) {
+      throw Error("Environment name must be specified");
+    }
+    if (protectedEnvironment(name)) {
+      throw Error(`Environment '${name}' is protected`);
+    }
+    provisionEnvironment(name, projects).catch(error => {
+      throw error;
+    });
+});
+
+```
+
+In the same directory add `payload.json` file with this content:
+
+```json
+{"name": "bob"}
+```
+
+Let's run our workflow with `brig`:
+
+```console
+$ brig run brigadecore/brigade-tutorial-config -f brigade.js -p payload.json --namespace brigade
+```
+
+Command above will trigger brigade workflow by executing `brigade.js` script with a data from `payload.json`.
+
+At this point our `brigade.js` script will simply use Kubernetes API to create new namespace. Name of new namespace is set in payload json.
+
+We first comparing it with a list of `protected` namespace names and then we're using 
```

Comment:
Wondering if this sentence can be re-worded?  Including, `We first comparing it` to something like `First, we compare the namespace...`, making it evident what `it` is referring to.